src/index.ts

```diff
@@ -549,8 +549,14 @@ function propertyConditions(
   const { debug } = options
   const propertyName = property.name
 
-  const varName = `${objName}.${propertyName}`
-  const propertyPath = `${path}.${propertyName}`
+  const identifier = isIdentifier(propertyName)
+  const varName = identifier
+    ? `${objName}.${propertyName}`
+    : `${objName}[${propertyName}]`
```

Comment:
I noticed here that "propertyName" doesn't require quotes around it... Are you sure you couldn't simply do:
```ts
const identifier = propertyName[0] === '"'
```
and skip the big regex?

FWIW it's completely legal to do `object.if` in modern JS.

Reply:
>Does this close #100 and #99?
it should :) 

> const identifier = propertyName[0] === '"'
yeah I think this should also work :)